runtime/vm/description.c

```diff
@@ -151,18 +151,22 @@ calculateInstanceDescription( J9VMThread *vmThread, J9Class *ramClass, J9Class *
 	 * set the corresponding description bit for each 
 	 */
 	{
+		BOOLEAN deepScanEnabled =  J9_ARE_ALL_BITS_SET(((vmThread)->javaVM)->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_DEEPSCAN);
```

Comment:
I'd like to see this rebased on #6489 as both are determining whether to perform the optimization or not - in the case of #6489 it's checking if the parent class has already done it and here it's checking for the option.

I think the two can be made into a single "shouldApplyOptimization" check that would be cleaner than having multiple checks.

Reply:
Added a single check `shouldSaveSelfReferencingFields which gets

1. Initialized by checking deep scan runtime option flag when declared 
```
BOOLEAN shouldSaveSelfReferencingFields = J9_ARE_ALL_BITS_SET(((vmThread)->javaVM)->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_DEEPSCAN); 
```
2. Reinitialized if a parent class exists

```/* Copy superclass description */
if (ramSuperClass) {
  ...
  /* If there are self referencing field offsets being inherited then self referencing fields of this class should be ignored.*/
  shouldSaveSelfReferencingFields = (shouldSaveSelfReferencingFields && (ramSuperClass->selfReferencingField1 == 0)) ? TRUE : FALSE;
}
```